### PR TITLE
Updated ff_prep with the ff_prep_refactored and renamed the files. ff…

### PR DESCRIPTION
**Pull request by maasva**
…_prep_refactored test removed.

### Description

Updated ff_prep with the refactored version. Removed the test that tested whether the outputs were the same as that is no longer relevant. 

### Requirements

Improving the code of ff_prep

### Configurations

NA

### Related Issues

#6

### CODE REVIEW

**Review by aminbksi — APPROVED** [review]
looks good, just make sure the package runs correctly and as intended locally

File: R/ff_prep.R [hunk 1]
```diff
@@ -62,280 +81,432 @@
...
-    stop("unknown what to process since no tiles or country or shape were given")
+
+  # Check input parameters validity
+  if (!hasvalue(tiles) && !hasvalue(country) && !hasvalue(shape)) {
+    stop("Unknown what to process since no tiles, country, or shape were given")
   }
-  if (hasvalue(shape) & !class(shape) == "SpatVector") {
-    stop("shape should be of class spatVector")
+  if (hasvalue(shape)) {
+    check_spatvector(shape)
```
[the remark this answers is not in the document]
Comment (maasva) [reply R/ff_prep.R]: Fixed

File: R/ff_prep.R [hunk 2]
```diff
@@ -2,42 +2,61 @@
...
+#' If `label_threshold = NA`, the ground truth will represent the total number of
+#' deforestation events rather than a binary label.
+#' @param add_xy Boolean indicating whether to add x and y coordinates as features. Default is FALSE.
 #'
 #' @return A list containing:
-#'   \item{data_matrix}{A list with features and labels for training}
+#'   \item{feature_dataset}{A list with features and labels for training}
 #'   \item{validation_matrix}{A list with features and labels for validation (if validation_sample > 0)}
 #'   \item{testindices}{Indices of the filtered samples}
 #'   \item{groundtruthraster}{A SpatRaster of the ground truth}
```
[the remark this answers is not in the document]
Comment (maasva) [reply R/ff_prep.R]: Fixed

File: R/ff_prep.R [hunk 3]
```diff
@@ -2,42 +2,61 @@
...
+#' Default is 1, meaning a pixel is labeled as deforested if at least one deforestation event has occurred.
+#' If `label_threshold = NA`, the ground truth will represent the total number of
+#' deforestation events rather than a binary label.
+#' @param add_xy Boolean indicating whether to add x and y coordinates as features. Default is FALSE.
 #'
 #' @return A list containing:
-#'   \item{data_matrix}{A list with features and labels for training}
+#'   \item{feature_dataset}{A list with features and labels for training}
 #'   \item{validation_matrix}{A list with features and labels for validation (if validation_sample > 0)}
 #'   \item{testindices}{Indices of the filtered samples}
```
[the remark this answers is not in the document]
Comment (maasva) [reply R/ff_prep.R]: fixed

**Review by jduijvenbode-WWF — APPROVED** [review]

File: config.yml [hunk 4]
```diff
@@ -7,4 +7,4 @@ DATA_FOLDER: "" #Should be empty on develop only change it locally.
 TEST:
   DATA_FOLDER: "tests/test_data"
   FF_PREP_QC_DATE: "2021-01-01"
-  FF_PREP_COUNTRY: "BRN"
+  FF_PREP_COUNTRY: "BDI"
```
Comment (aminbksi) [inline config.yml:10]: should be BRN

**Review by aminbksi — APPROVED** [review]

**Pull request merged**